app/main/views/agreements.py

```diff
@@ -66,14 +67,13 @@ def create_framework_agreement():
     db.session.add(audit_event)
     db.session.commit()
 
-    return jsonify(agreement=framework_agreement.serialize()), 201
+    return single_result_response("agreement", framework_agreement), 201
```

Comment:
It would be good to have `"agreement"` as a constant (same with other places where we pass in the name of the resource).